datadog/provider.go

```diff
@@ -104,5 +134,12 @@ func translateClientError(err error, msg string) error {
 		msg = "an error occurred"
 	}
 
+	if errBody, ok := err.(datadog.GenericOpenAPIError); ok {
+		return fmt.Errorf(msg+" (datadog.GenericOpenAPIError): %s", string(errBody.Body()))
```

Comment:
Let's still print `err.Error()` in that case too, it at least gives us the status code, and there can be cases when a `datadog.GenericOpenAPIError` can be returned before any request is made, so it won't have a `Body`